docs/resources/servicestage_repository_authorization.md

```diff
@@ -0,0 +1,112 @@
+---
+subcategory: "ServiceStage"
+---
+
+# huaweicloud_servicestage_repository_authorization
+
+This resource is used for the ServiceStage service to establish the authorization relationship with various types of
+the Open-Source repository.
+
+## Example Usage
+
+### Create the authorization using personal access token
+
+```hcl
+variable "authorization_name"
+variable "repository_host"
+variable "persional_access_token"
+
+resource "huaweicloud_servicestage_repository_authorization" "test" {
+  type  = "github"
+  name  = var.authorization_name
+  host  = var.repository_host
+  token = var.personal_access_token
+}
+```
+
+### Create the authorization using user name and password
+
+```hcl
+variable "authorization_name"
+variable "domain_name"
+variable "tenant_name"
+variable "password"
+
+resource "huaweicloud_servicestage_repository_authorization" "test" {
+  type      = "devcloud"
+  name      = var.authorization_name
+  user_name = format("%s/%s", var.domain_name, var.tenant_name)
+  password  = var.password
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specified the region in which to create the repository authorization.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new authorization.
+
+* `name` - (Required, String, ForceNew) Specified the authorization name. The name can contain of 4 to 63 characters,
+  only letters, digits, underscores (_), hyphens (-) and dots (.) are allowed.
+  Changing this parameter will create a new authorization.
+
+* `type` - (Required, String, ForceNew) Specified the authorization type. The valid values are as follows:
+  + **devcloud**
+  + **bitbucket**
+  + **github**
+  + **gitlab**
+  + **gitee**
+
+  Changing this parameter will create a new authorization.
+
+* `host` - (Optional, String, ForceNew) Specified the host name of the repository.
+  Required if the `type` is **github**, **gitlab** or **gitee**. This parameter and `user_name` are alternative.
+  Changing this parameter will create a new authorization.
+
+* `token` - (Optional, String, ForceNew) Specified the personal access token of the repository.
+  Required if the `host` is provided.
+  Changing this parameter will create a new authorization.
+
+* `user_name` - (Optional, String, ForceNew) Specified the user name of the repository.
```

Comment:
instead of make such parameters different for different types, how about just split this repository_authorization into personal token autheorization and password autorization?

Reply:
done.
I separate two resources: `huaweicloud_servicestage_repo_token_authorization` and `huaweicloud_servicestage_repo_pwd_authorization`.